src/client/lazy-app/Compress/Modal/index.tsx

```diff
@@ -0,0 +1,148 @@
+import { h, Component } from 'preact';
+import * as style from './style.css';
+import 'add-css:./style.css';
+import { linkRef } from 'shared/prerendered-app/util';
+import {
+  InfoIcon,
+  DiamondStarIcon,
+  ModalErrorIcon,
+} from 'client/lazy-app/icons';
+
+interface Props {}
+
+export interface ModalMessage {
+  type: 'info' | 'error' | 'update';
+  title: string;
+  content: string;
+}
+
+interface State {
+  message: ModalMessage;
+  shown: boolean;
+}
+
+export default class Modal extends Component<Props, State> {
+  state: State = {
+    message: {
+      type: 'info',
+      title: 'default',
+      content: 'default',
+    },
+    shown: false,
+  };
+
+  private modal?: HTMLElement;
+  private returnFocusElement?: HTMLElement | null;
+
+  showModal(message: ModalMessage) {
+    if (this.state.shown) return;
+    if (!this.modal) return;
+
+    // Set element to return focus to after hiding
+    this.returnFocusElement = document.activeElement as HTMLElement;
+
+    this.modal.style.display = '';
+    this.setState({
+      message: message,
+      shown: true,
+    });
+    // Wait for the 'display' reset to take place, then focus
+    setTimeout(() => {
+      this.modal?.querySelector('button')?.focus();
+    }, 0);
+  }
+
+  hideModal() {
+    this.setState({
+      message: { ...this.state.message },
+      shown: false,
+    });
+    setTimeout(() => {
+      this.modal && (this.modal.style.display = 'none');
+      this.returnFocusElement?.focus();
+    }, 250);
+  }
+
+  private _getCloseButton() {
+    return this.modal!.querySelector('button')!;
```

Comment:
We should have a dedicated class or id for the close button. Feels brittle otherwise with user-provided content for the modal.

Reply:
That was just lazy coding on my part LOL, I'll make a dedicated id for it

Reply:
Also seems odd to have a method like this, but then other code above getting the button manually. But I don't think we'll need this at all with `<dialog>`.